### ISSUE TITLE
Generate `fstar` on CI

### ISSUE THREAD

**Issue by geonnave** [openwsn-berkeley/edhoc-rs#57]
Here is a starting point (note that it sill fails as per [this failed job](https://github.com/openwsn-berkeley/edhoc-rs/actions/runs/4753145822/jobs/8444380595)):

```
  build-fstar:
    runs-on: ubuntu-latest
    permissions:
      contents: write
    steps:
      - name: Checkout code
        uses: actions/checkout@v3
        with:
          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}

      - name: Checkout hacspec-v2
        uses: actions/checkout@v3
        with:
          repository: 'hacspec/hacspec-v2'
          path: hacspec-v2-repo

      - uses: ocaml/setup-ocaml@v2
        with:
          ocaml-compiler: 4.14.x

      - uses: dtolnay/rust-toolchain@1.67 # TODO: change this to `stable` when we update nightly

      - run: cargo install --path cli/driver && cargo install --path cli/subcommands

      - run: opam install . --deps-only
        working-directory: hacspec-v2-repo/engine

      - run: opam exec -- dune build
        working-directory: hacspec-v2-repo/engine

      - run: |
          sed -i -E "s/(edhoc-crypto.*), default-features = false(.*)/\1\2/" Cargo.toml
          cargo circus fstar
        working-directory: hacspec
```

**Comment by geonnave** [comment]
This is ongoing on #59.

Can build hacspec-v2 docker image, however the call to `cargo-circus` still fails. See [here](https://github.com/openwsn-berkeley/edhoc-rs/actions/runs/4766606455/jobs/8473835561).

**Comment by geonnave** [comment]
Solved in #81
